src/YieldDisburser.sol

```diff
@@ -188,16 +226,36 @@ contract YieldDisburser is OwnableUpgradeable {
         lastClaimedBlocknumber = _lastClaimedBlocknumber;
     }
 
-    function addProject(address projectAddress) public onlyOwner {
-        breadchainProjects.push(projectAddress);
+    function queueProjectAddition(address project) public onlyOwner {
+        for (uint256 i; i < queuedProjectsForAddition.length; ++i) {
+            if (queuedProjectsForAddition[i] == project) {
+                revert ProjectNotFound();
+            }
+        }
+        for (uint256 i; i < breadchainProjects.length; ++i) {
+            if (breadchainProjects[i] == project) {
+                revert ProjectExistsOrAlreadyQueued();
+            }
+        }
+        queuedProjectsForAddition.push(project);
     }
 
-    function removeProject(address projectAddress) public onlyOwner {
-        for (uint256 i = 0; i < breadchainProjects.length; i++) {
-            if (breadchainProjects[i] == projectAddress) {
-                delete breadchainProjects[i];
-                break;
+    function queueProjectRemoval(address project) public onlyOwner {
+        for (uint256 i; i < queuedProjectsForRemoval.length; ++i) {
+            if (queuedProjectsForRemoval[i] == project) {
+                revert ProjectExistsOrAlreadyQueued();
+            }
+        }
+        for (uint256 i; i < breadchainProjects.length; ++i) {
+            if (breadchainProjects[i] == project) {
+                queuedProjectsForRemoval.push(project);
+                return;
             }
         }
+        revert ProjectNotFound();
+    }
```

Comment:
Here I'd use the same error for when a project is already in the queue as in `queueProjectAddition()`. 

Again I'd check to see if the project is actually an existing member before checking the queue for the same reason.

Also the logic in this method follows a different pattern to  `queueProjectAddition()`. I'd do the checks first and push the address to the array afterwards like in the other method.

Reply:
should be addressed with fbeb773